src/amo/components/HeroRecommendation/index.js

```diff
@@ -174,31 +176,31 @@ export class HeroRecommendationBase extends React.Component<InternalProps> {
             )}
             <div className="HeroRecommendation-info">
               <div className="HeroRecommendation-recommended">
-                {shelfData ? (
+                {loading ? (
```

Comment:
using an explicit `loading` variable is more explicit and probably better... but it wasn't necessary.  We can check `shelfData` as it is currently - that part worked fine, it was the testing the individual data (`summary`, etc) inside `shelfData` that introduced the bug.  

It wouldn't have needed the logic reversing for each of them either.

Reply:
I wasn't really happy with the way it was coded, preferring to have the logic to decide if we're in loading mode in one place, and then using that elsewhere. It's true that it did result in more changes, but I think the code overall is better this way.